tests/integration/integration.go

```diff
@@ -120,6 +138,92 @@ func K3sServerArgs() []string {
 	return args
 }
 
+// K3sDefaultDeployments checks if the default deployments for K3s are ready, otherwise returns an error
+func K3sDefaultDeployments() error {
+	return CheckDeployments([]string{"coredns", "local-path-provisioner", "metrics-server", "traefik"})
+}
+
+// CheckDeployments checks if the provided list of deployments are ready, otherwise returns an error
+func CheckDeployments(deployments []string) error {
+
+	deploymentSet := make(map[string]bool)
+	for _, d := range deployments {
+		deploymentSet[d] = false
+	}
+	res, err := K3sCmd("kubectl get deployments --no-headers -A")
+	if err != nil {
+		return err
+	}
+	res = strings.TrimSpace(res)
+
+	split := strings.Split(res, "\n")
+	for _, rec := range split {
+		fields := strings.Fields(rec)
+		if _, ok := deploymentSet[fields[1]]; ok && fields[2] == "1/1" {
+			deploymentSet[fields[1]] = true
+		}
+	}
+	for d, found := range deploymentSet {
+		if !found {
+			return fmt.Errorf("failed to deploy %s\n%s", d, res)
+		}
+	}
+	return nil
+}
+
+func ParsePods() ([]Pod, error) {
```

Comment:
Is now maybe a good time to come up with some helpers to create an actual Kubernetes client and just get the nodes and pods directly, instead of wrapping shell commands and parsing the output into our own Pod/Node structs?

Reply:
Resolved